homeassistant/components/weather/meteoswiss.py

```diff
@@ -0,0 +1,188 @@
+"""
+Support for MeteoSwiss.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/weather.meteoswiss/
+"""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.weather import (
+    ATTR_FORECAST_CONDITION, ATTR_FORECAST_TEMP, ATTR_FORECAST_TEMP_LOW,
+    ATTR_FORECAST_TIME, ATTR_FORECAST_PRECIPITATION, PLATFORM_SCHEMA,
+    WeatherEntity)
+from homeassistant.const import CONF_NAME, STATE_UNKNOWN, TEMP_CELSIUS
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_CONDITION = 'meteoswiss_condition'
+
+ATTRIBUTION = "Data provided by MeteoSwiss"
+
+
+CONF_POSTAL_CODE = 'postal_code'
+
+DEFAULT_NAME = 'MeteoSwiss'
+API_URL = 'https://app-prod-ws.meteoswiss-app.ch/v1/plzDetail?plz={}'
+
+SCAN_INTERVAL = timedelta(minutes=10)
+
+CONDITION_CLASSES = {
+    'clear-night': [101, 102, 126],
+    'cloudy': [5, 105, 35, 135],
+    'fog': [27, 127, 28, 128],
+    'hail': [],
+    'lightning': [12, 112],
+    'lightning-rainy': [13, 113, 23, 123, 24, 124, 25, 125],
+    'partlycloudy': [3, 103, 4, 104],
+    'pouring': [17, 117, 20, 120, 33, 133],
+    'rainy': [6, 106, 9, 109, 14, 114, 29, 129, 32, 132],
+    'snowy': [8, 108, 11, 111, 16, 116, 19, 119, 22, 122, 30, 130, 34, 134],
+    'snowy-rainy': [7, 107, 10, 110, 15, 115, 18, 118, 21, 121, 31, 131],
+    'sunny': [1, 2, 26],
+    'windy': [],
+    'windy-variant': [],
+    'exceptional': [],
+}
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_POSTAL_CODE): vol.Range(min=1000, max=999999),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the MeteoSwiss platform."""
+    postal_code = str(config.get(CONF_POSTAL_CODE))
+    if len(postal_code) == 4:
```

Comment:
It feels like that most people will use four digits for the postal code thus add `00` in all cases and let the validator check for four digits.